src/Servers/Kestrel/Core/src/ListenOptionsHttpsExtensions.cs

```diff
@@ -163,33 +163,28 @@ public static ListenOptions UseHttps(this ListenOptions listenOptions, Action<Ht
     {
         ArgumentNullException.ThrowIfNull(configureOptions);
 
+        // TODO (acasey): apply to other overloads?
+        listenOptions.KestrelServerOptions.EnableTlsConfigurationLoading();
+
         var options = new HttpsConnectionAdapterOptions();
+
+        // This will throw if there are configured defaults and defaults are not supported (i.e. in slim scenarios)
         listenOptions.KestrelServerOptions.ApplyHttpsDefaults(options);
+
         configureOptions(options);
-        listenOptions.KestrelServerOptions.ApplyDefaultCertificate(options);
 
-        if (options.ServerCertificate == null && options.ServerCertificateSelector == null)
+        if (!options.HasServerCertificateOrSelector)
         {
-            throw new InvalidOperationException(CoreStrings.NoCertSpecifiedNoDevelopmentCertificateFound);
+            // This will throw if default certificates are not enabled (i.e. in slim scenarios)
```

Comment:
```suggestion
```

This shouldn't throw now that we're calling `EnableTlsConfigurationLoading()`, right?

Reply:
Yes, I'll have to review all the comments if we go ahead with this model.

Reply:
I think I caught all of them.